src/App.jsx

```diff
@@ -14,33 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const sortAlphabetical = 'name';
+const sortLength = 'length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isChanged, setIsChanged] = useState(false);
+
+
+  let sortedListGoods = getPrepareGoods(goodsFromServer, sortField, isReversed);
+
+  function getPrepareGoods(goods, sortField, isReversed) {
+    let SortedGoods = [...goods];
+    if (sortField) {
+      SortedGoods.sort((good1, good2) => {
+        switch (sortField) {
+          case sortAlphabetical:
+            return good1.localeCompare(good2);
+          case sortLength:
+            return good1.length - good2.length;
+          default:
+            return 0;
+        }
+      });
+    }
+    if (isReversed) {
+      SortedGoods.reverse();
+    }
+    return SortedGoods;
+  }
+
+  const handleReverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(sortAlphabetical)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== sortAlphabetical,
+          })}
+        >
+          Sort alphabetically
+        </button>
+        <button
+          onClick={() => setSortField(sortLength)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== sortLength,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={handleReverse}
+          type="button"
+          className={cn('button is-warning', { 'is-light': !isReversed })}
+        >
+          Reverse
+        </button>
+        {(sortField !== "" || isReversed) && (
```

Comment:
```suggestion
        {(sortField || isReversed) && (
```
can be shorter in this case